src/solver/opamSolver.ml

```diff
@@ -772,18 +772,27 @@ let dump_universe universe oc =
 
 let filter_solution filter t =
   let t = OpamCudf.ActionGraph.copy t in
-  let rec rm iter_deps v =
-    if OpamCudf.ActionGraph.mem_vertex t v then (
-      iter_deps (rm iter_deps) t v;
-      OpamCudf.ActionGraph.remove_vertex t v
-    ) in
-  OpamCudf.ActionGraph.iter_vertex
-    (function
+  let rec rm ?(not_reinstall=fun _ -> true) iter_deps v =
```

Comment:
an aside only, it may be nice to use the positive case convention. `?(reinstall=fun _ -> false)`, that way readers don't need to process the double negative to comprehend the positive result.

Reply:
good point. Done. Thanks!